vita3k/gui/src/information_bar.cpp

```diff
@@ -37,31 +37,57 @@
 
 namespace gui {
 
-struct NoticeInfo {
+struct NoticeList {
     std::string id;
     std::string content_id;
     std::string group;
     std::string type;
     time_t time;
-    std::string name;
-    std::string msg;
+    bool is_new;
 };
 
-struct NoticeList {
-    std::string id;
-    std::string content_id;
-    std::string group;
-    std::string type;
-    time_t time;
+struct NoticeInfo : NoticeList {
+    std::string name;
+    std::string msg;
 };
 
 static std::map<std::string, std::vector<NoticeList>> notice_list;
 static std::map<std::string, int> notice_list_count_new;
-static std::map<std::string, std::map<time_t, bool>> notice_list_new;
-static std::map<time_t, bool> notice_info_new;
 static int notice_info_count_new = 0;
 static std::vector<NoticeInfo> notice_info;
 
+void erase_app_notice(GuiState &gui, const std::string &title_id) {
+    auto &notice_global = notice_list["global"];
+
+    // Check if the notice list is empty
+    if (notice_global.empty()) {
+        LOG_WARN("Notice list is empty.");
+        return;
+    }
+
+    auto notice_list_it = notice_global.begin();
+    while (notice_list_it != notice_global.end()) {
+        if (notice_list_it->id == title_id) {
+            // Find and erase the corresponding entry in notice_info
+            const auto notice_info_it = std::find_if(notice_info.begin(), notice_info.end(), [&](const NoticeInfo &n) {
+                return n.time == notice_list_it->time;
+            });
+            if (notice_info_it != notice_info.end())
+                notice_info.erase(notice_info_it); // Erase the entry from notice_info
+
+            // Erase the entry from notice_info_icon if it exists
+            if (gui.notice_info_icon.contains(notice_list_it->time))
+                gui.notice_info_icon.erase(notice_list_it->time);
```

Comment:
```suggestion
                gui.notice_info_icon.erase(notice_list_it->time);
```

Reply:
no get crash if element is missing on map ?